### PR TITLE
remove set_leader from cluster_info

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 replay_stage is still hard to read, partly because of deprecated cluster_info.leader_data()

 #### Summary of Changes
 * remove deprecated API
 * move RPC to bank APIs for current leader (which matches rpc.get_slot())
 * clean up replay_stage some more

Fixes #

### CODE REVIEW

File: core/src/replay_stage.rs [hunk 1]
```diff
@@ -242,33 +247,30 @@ impl ReplayStage {
...
-                    );
-                    debug!(
-                        "poh_recorder new working bank: me: {} next_slot: {} next_leader: {}",
-                        my_pubkey,
-                        tpu_bank.slot(),
-                        next_leader
-                    );
-                    poh_recorder.lock().unwrap().set_bank(&tpu_bank);
-                }
+            // I guess I missed my slot
```
Comment (mvines) [inline core/src/replay_stage.rs]: Does this deserve a log or datapoint?

File: core/src/cluster_info.rs [hunk 2]
```diff
@@ -78,9 +78,6 @@ pub struct ClusterInfo {
     pub gossip: CrdsGossip,
     /// set the keypair that will be used to sign crds values generated. It is unset only in tests.
     pub(crate) keypair: Arc<Keypair>,
-    // TODO: remove gossip_leader_pubkey once all usage of `set_leader()` and `leader_data()` is
```
Comment (mvines) [inline core/src/cluster_info.rs]: Finally!!1!

**Comment by mvines** [comment]
heh, you have 1 minute to review.  GO!
image

File: core/src/replay_stage.rs (hunk 1 above)
Comment (rob-solana) [reply core/src/replay_stage.rs]: there's a log above, could be changed to info!

I think this is somewhat rare (used to happen in local_cluster tests often, though) and harmless.

**Pull request merged**